libs/entitlement/src/main/java/org/elasticsearch/entitlement/runtime/policy/entitlements/FilesEntitlement.java

```diff
@@ -52,18 +162,41 @@ public static FilesEntitlement build(List<Object> paths) {
         List<FileData> filesData = new ArrayList<>();
         for (Object object : paths) {
             Map<String, String> file = new HashMap<>((Map<String, String>) object);
-            String path = file.remove("path");
-            if (path == null) {
-                throw new PolicyValidationException("files entitlement must contain path for every listed file");
-            }
+            String pathAsString = file.remove("path");
+            String relativePathAsString = file.remove("relative_path");
+            String relativeTo = file.remove("relative_to");
             String mode = file.remove("mode");
+
+            if (file.isEmpty() == false) {
+                throw new PolicyValidationException("unknown key(s) [" + file + "] in a listed file for files entitlement");
```

Comment:
@jdconrad I noticed in these policy validation exceptions we end up losing context. Although the parser will give a location in the file, it won't be the specific element within the files entitlement array. I wonder if we could improve this by having a callback that parses each element, so that the parser could still give the exact location where a problem exists.